api/python/cell_census/src/cell_census/experiment_query/axis.py

```diff
@@ -6,10 +6,9 @@
 import pyarrow as pa
 from typing_extensions import TypedDict
 
-# Type declaration/helpers local to this file
-#
-Coordinates = Tuple[Union[slice, int, npt.ArrayLike], ...]
-ValueFilter = str
+AxisCoordinate = Union[slice, int, npt.ArrayLike]
+AxisCoordinates = Tuple[AxisCoordinate, ...]
+AxisValueFilter = str
```

Comment:
When would a user want to grab a _slice_ of `joinid`s? Does the order of `joinid`s ever have significance? Sure someone might want to grab `[0:10]` for  playing around with small subset of an axis, but if that's not a realistic use, why support it at this API level? `list(range(10))` is an easy enough workaround, if needed.

(Even if we change joinids from being ints to strs for `var`, the ordering doesn't seem to have significance.)

If we do away with slices, we can simplify to:
```
AxisCoordinate = Union[int, str, bytes]
AxisCoordinates = List[AxisCoordinate]
```
And now these type names are more accurate.  The singular form `AxisCoordinate` is now _always_ a scalar value, and `AxisCoordinates` can never be a list of lists, which is a case that makes the current name also misleading.

Reply:
I think you are right, at least as the census currently uses the joinids (ie. obs is _only_ indexed on the joinid - which is a census schema assumption, not a SOMA assumption).  I didn't want the API to make too many assumptions which would restrict our ability to evolve the census schema, which is why I exposed the full set of underlying capabilities.

I think we can do either way:
* minimal API surface area that would have to change if we evolve the schema
* or, more flexible API surface area that would be less likely to change

probably simplest to discuss IRT